postform_decoder/src/lib.rs

```diff
@@ -258,6 +279,28 @@ impl ElfMetadata {
         }
     }
 
+    fn get_str_ptr<'a>(&self, message: &'a [u8]) -> Result<(u64, &'a [u8]), Error> {
+        if self.platform_descriptors.ptr_size as usize == std::mem::size_of::<u32>() {
```

Comment:
This is pretty awful, need to refactor it an make it generic for the size of the pointer.